### ISSUE TITLE
JSON: Sort document does not work

### ISSUE THREAD

**Issue by hediet** [microsoft/vscode#178229]
This error is logged:
```
 ERR Request json/sort failed with message: Range#create called with invalid arguments[[object Object], [object Object], undefined, undefined]: Error: Request json/sort failed with message: Range#create called with invalid arguments[[object Object], [object Object], undefined, undefined]
    at m.h (c:\Users\hdieterichs\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:98:126151)
```

This is the JSON document:
```json
{
    "z": true,
    "a": {
        "a": "value",
    }
}
```

It seems to be caused by the trailing comma after `"value"`

**Comment by aiday-mar** [comment]
Hi thanks for filing the issue. Indeed currently trailing commas are not supported. The files must be formatted correctly before applying the JSON sorting. This is a point on my TODO list.

**Comment by Vipank** [comment]
Can I work on this issue?

**Comment by Vipank** [comment]
Thanks @aiday-mar, I'll try to understand the code. This will be my first open-source contribution. Will let you know my approach.

**Comment by rchl** [comment]
As I've mentioned in the duplicate, a trailing comma works fine in another case like:

```
{
    "foo": {"a": "b"},
}
```

Saying that in relation to the comment that trailing commas are not currently supported.

**Comment by a-gn** [comment]
Could the error message mention the nature and location of the error? That would help with removing the training comma in a large file, until this is supported.

**Comment by denisw** [comment]
I started looking into this as I would like to sort my `settings.json` file with this feature. I cannot promise that I'll find the time to actually create a fix, but I want to at least share what I found out so far.

The error seems to be thrown by [this line](https://github.com/microsoft/vscode-json-languageservice/blob/bbba2d576f1983edee03c088a4d6bea70852d422/src/utils/sort.ts#L343) in the `src/utils/sort.ts` file of `vscode-json-languageservice`: 

```ts
const range: Range = Range.create(Position.create(propertyTree.beginningLineNumber!, 0), Position.create(propertyTree.endLineNumber! + 1, 0));
```

Further debugging revealed that when a document contains a nested object with a trailing comma (e.g. `{ "b: { "x": 1 }, "a": 2 }`), there is a situation where `propertyTree.endLineNumber` is `undefined`, which in turn causes `Range.create()` to throw due to an invalid end position.

The question is how this `PropertyTree` without `endLineNumber` came to be. I assume that it is a bug in `findPropertyTree`, which seems to construct a tree of `PropertyTree` objects from the input JSONC string, but I still need to dig further. In case I don't find the time to continue, though, I hope that this helps someone else to make sense of the issue.
